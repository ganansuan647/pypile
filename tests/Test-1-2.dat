[contral]
2
end;
[arrange]
24 0
-5.0 2.5
-5.0 -2.5
-5.0 7.5
-5.0 -7.5
-5.0 12.5
-5.0 -12.5
-5.0 17.5
-5.0 -17.5
0 2.5
0 -2.5
0 7.5
0 -7.5
0 12.5
0 -12.5
0 17.5
0 -17.5
5.0 2.5
5.0 -2.5
5.0 7.5
5.0 -7.5
5.0 12.5
5.0 -12.5
5.0 17.5
5.0 -17.5
end;
[no_simu]
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
<0>
0 1 0 0 1
1 3.9 2.0 4
10 9.5 2.0 5000 18.0 10
 9.0 2.0 3000 13.7 9
 29.1 2.0 5000 15.5 30
 5.8 2.0 5000 30.0 6
 2.8 2.0 5000 17.0 3
 2.4 2.0 10000 18.0 3
 3.0 2.0 10000 17.0 4
 4.0 2.0 5000 17.0 4
 14.9 2.0 5000 31.0 15
 5.6 2.0 10000 32.0 6
10000 3e7 1
end
[simu_pe]
end;