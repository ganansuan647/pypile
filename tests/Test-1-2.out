-----------------------------------------------------------------------------------------------------------------------------------------------------
|           _____                _____                            _____                    _____                    _____            _____           |
|          /\    \              |\    \                          /\    \                  /\    \                  /\    \          /\    \          |
|         /::\    \             |:\____\                        /::\    \                /::\    \                /::\____\        /::\    \         |
|        /::::\    \            |::|   |                       /::::\    \               \:::\    \              /:::/    /       /::::\    \        |
|       /::::::\    \           |::|   |                      /::::::\    \               \:::\    \            /:::/    /       /::::::\    \       |
|      /:::/\:::\    \          |::|   |                     /:::/\:::\    \               \:::\    \          /:::/    /       /:::/\:::\    \      |
|     /:::/__\:::\    \         |::|   |                    /:::/__\:::\    \               \:::\    \        /:::/    /       /:::/__\:::\    \     |
|    /::::\   \:::\    \        |::|   |                   /::::\   \:::\    \              /::::\    \      /:::/    /       /::::\   \:::\    \    |
|   /::::::\   \:::\    \       |::|___|______            /::::::\   \:::\    \    ____    /::::::\    \    /:::/    /       /::::::\   \:::\    \   |
|  /:::/\:::\   \:::\____\      /::::::::\    \          /:::/\:::\   \:::\____\  /\   \  /:::/\:::\    \  /:::/    /       /:::/\:::\   \:::\    \  |
| /:::/  \:::\   \:::|    |    /::::::::::\____\        /:::/  \:::\   \:::|    |/::\   \/:::/  \:::\____\/:::/____/       /:::/__\:::\   \:::\____\ |
| \::/    \:::\  /:::|____|   /:::/~~~~/~~              \::/    \:::\  /:::|____|\:::\  /:::/    \::/    /\:::\    \       \:::\   \:::\   \::/    / |
|  \/_____/\:::\/:::/    /   /:::/    /                  \/_____/\:::\/:::/    /  \:::\/:::/    / \/____/  \:::\    \       \:::\   \:::\   \/____/  |
|           \::::::/    /   /:::/    /                            \::::::/    /    \::::::/    /            \:::\    \       \:::\   \:::\    \      |
|            \::::/    /   /:::/    /                              \::::/    /      \::::/____/              \:::\    \       \:::\   \:::\____\     |
|             \::/____/    \::/    /                                \::/____/        \:::\    \               \:::\    \       \:::\   \::/    /     |
|              ~~           \/____/                                  ~~               \:::\    \               \:::\    \       \:::\   \/____/      |
|                                                                                      \:::\    \               \:::\    \       \:::\    \          |
|                                                                                       \:::\____\               \:::\____\       \:::\____\         |
|                                                                                        \::/    /                \::/    /        \::/    /         |
|                                                                                         \/____/                  \/____/          \/____/          |
|                                                                                                                                                    |
|                                                                                                                                                    |
|                                                                                                                                Version:1.0.2, 2025 |
|                                                                                                                                    By: Lingyun Gou |
-----------------------------------------------------------------------------------------------------------------------------------------------------
# 截面刚度矩阵报告

+-------+------------+-------------+------------+-------------+-----------+------------+
| DOF   |         Ux |          Uy |         Uz |          Rx |        Ry |         Rz |
+=======+============+=============+============+=============+===========+============+
| Ux    | 3.7536e+06 |  0          | 0          |  0          | 1.651e+07 | 0          |
+-------+------------+-------------+------------+-------------+-----------+------------+
| Uy    | 0          |  3.6852e+06 | 0          | -1.6309e+07 | 0         | 0          |
+-------+------------+-------------+------------+-------------+-----------+------------+
| Uz    | 0          |  0          | 3.4059e+07 |  0          | 0         | 0          |
+-------+------------+-------------+------------+-------------+-----------+------------+
| Rx    | 0          | -1.6273e+07 | 0          |  4.6447e+09 | 0         | 0          |
+-------+------------+-------------+------------+-------------+-----------+------------+
| Ry    | 1.6474e+07 |  0          | 0          |  0          | 5.77e+08  | 0          |
+-------+------------+-------------+------------+-------------+-----------+------------+
| Rz    | 0          |  0          | 0          |  0          | 0         | 5.7217e+08 |
+-------+------------+-------------+------------+-------------+-----------+------------+

## 单位说明
- 位移相关刚度: kN/m
- 转角相关刚度: kN·m/rad
- 注: 比最大值小10个数量级的值已视为0 (阈值: 4.6447e-01)
# 桩群计算结果综合报告

## 摘要
- 总桩数: 24根

## 桩顶位移和内力

### 位移和转角
+------------+--------+---------+---------+---------+----------+-----------+-----------+-----------+
|   Pile No. |   X(m) |    Y(m) |   Ux(m) |   Uy(m) |    Uz(m) |   Sx(rad) |   Sy(rad) |   Sz(rad) |
+============+========+=========+=========+=========+==========+===========+===========+===========+
|          0 |   -4.5 |   3.625 | 2.17493 |       0 | 0.794076 |         0 |  0.111215 |         0 |
+------------+--------+---------+---------+---------+----------+-----------+-----------+-----------+
|          1 |   -4.5 |  -3.625 | 2.17493 |       0 | 0.794076 |         0 |  0.111215 |         0 |
+------------+--------+---------+---------+---------+----------+-----------+-----------+-----------+
|          2 |   -4.5 |   8.125 | 2.17493 |       0 | 0.794076 |         0 |  0.111215 |         0 |
+------------+--------+---------+---------+---------+----------+-----------+-----------+-----------+
|          3 |   -4.5 |  -8.125 | 2.17493 |       0 | 0.794076 |         0 |  0.111215 |         0 |
+------------+--------+---------+---------+---------+----------+-----------+-----------+-----------+
|          4 |   -4.5 |  12.625 | 2.17493 |       0 | 0.794076 |         0 |  0.111215 |         0 |
+------------+--------+---------+---------+---------+----------+-----------+-----------+-----------+
|          5 |   -4.5 | -12.625 | 2.17493 |       0 | 0.794076 |         0 |  0.111215 |         0 |
+------------+--------+---------+---------+---------+----------+-----------+-----------+-----------+
|          6 |   -4.5 |  17.125 | 2.17493 |       0 | 0.794076 |         0 |  0.111215 |         0 |
+------------+--------+---------+---------+---------+----------+-----------+-----------+-----------+
|          7 |   -4.5 | -17.125 | 2.17493 |       0 | 0.794076 |         0 |  0.111215 |         0 |
+------------+--------+---------+---------+---------+----------+-----------+-----------+-----------+
|          8 |      0 |   3.625 | 2.17493 |       0 | 0.293608 |         0 |  0.111215 |         0 |
+------------+--------+---------+---------+---------+----------+-----------+-----------+-----------+
|          9 |      0 |  -3.625 | 2.17493 |       0 | 0.293608 |         0 |  0.111215 |         0 |
+------------+--------+---------+---------+---------+----------+-----------+-----------+-----------+
|         10 |      0 |   8.125 | 2.17493 |       0 | 0.293608 |         0 |  0.111215 |         0 |
+------------+--------+---------+---------+---------+----------+-----------+-----------+-----------+
|         11 |      0 |  -8.125 | 2.17493 |       0 | 0.293608 |         0 |  0.111215 |         0 |
+------------+--------+---------+---------+---------+----------+-----------+-----------+-----------+
|         12 |      0 |  12.625 | 2.17493 |       0 | 0.293608 |         0 |  0.111215 |         0 |
+------------+--------+---------+---------+---------+----------+-----------+-----------+-----------+
|         13 |      0 | -12.625 | 2.17493 |       0 | 0.293608 |         0 |  0.111215 |         0 |
+------------+--------+---------+---------+---------+----------+-----------+-----------+-----------+
|         14 |      0 |  17.125 | 2.17493 |       0 | 0.293608 |         0 |  0.111215 |         0 |
+------------+--------+---------+---------+---------+----------+-----------+-----------+-----------+
|         15 |      0 | -17.125 | 2.17493 |       0 | 0.293608 |         0 |  0.111215 |         0 |
+------------+--------+---------+---------+---------+----------+-----------+-----------+-----------+
|         16 |    4.5 |   3.625 | 2.17493 |       0 | -0.20686 |         0 |  0.111215 |         0 |
+------------+--------+---------+---------+---------+----------+-----------+-----------+-----------+
|         17 |    4.5 |  -3.625 | 2.17493 |       0 | -0.20686 |         0 |  0.111215 |         0 |
+------------+--------+---------+---------+---------+----------+-----------+-----------+-----------+
|         18 |    4.5 |   8.125 | 2.17493 |       0 | -0.20686 |         0 |  0.111215 |         0 |
+------------+--------+---------+---------+---------+----------+-----------+-----------+-----------+
|         19 |    4.5 |  -8.125 | 2.17493 |       0 | -0.20686 |         0 |  0.111215 |         0 |
+------------+--------+---------+---------+---------+----------+-----------+-----------+-----------+
|         20 |    4.5 |  12.625 | 2.17493 |       0 | -0.20686 |         0 |  0.111215 |         0 |
+------------+--------+---------+---------+---------+----------+-----------+-----------+-----------+
|         21 |    4.5 | -12.625 | 2.17493 |       0 | -0.20686 |         0 |  0.111215 |         0 |
+------------+--------+---------+---------+---------+----------+-----------+-----------+-----------+
|         22 |    4.5 |  17.125 | 2.17493 |       0 | -0.20686 |         0 |  0.111215 |         0 |
+------------+--------+---------+---------+---------+----------+-----------+-----------+-----------+
|         23 |    4.5 | -17.125 | 2.17493 |       0 | -0.20686 |         0 |  0.111215 |         0 |
+------------+--------+---------+---------+---------+----------+-----------+-----------+-----------+

### 力和转角
+------------+--------+---------+----------+----------+-------------+------------+-------------+------------+
|   Pile No. |   X(m) |    Y(m) |   Nx(kN) |   Ny(kN) |      Nz(kN) |   Mx(kN·m) |    My(kN·m) |   Mz(kN·m) |
+============+========+=========+==========+==========+=============+============+=============+============+
|          0 |   -4.5 |   3.625 |   416667 |        0 | 1.12689e+06 |          0 | 2.03598e+06 |          0 |
+------------+--------+---------+----------+----------+-------------+------------+-------------+------------+
|          1 |   -4.5 |  -3.625 |   416667 |        0 | 1.12689e+06 |          0 | 2.03598e+06 |          0 |
+------------+--------+---------+----------+----------+-------------+------------+-------------+------------+
|          2 |   -4.5 |   8.125 |   416667 |        0 | 1.12689e+06 |          0 | 2.03598e+06 |          0 |
+------------+--------+---------+----------+----------+-------------+------------+-------------+------------+
|          3 |   -4.5 |  -8.125 |   416667 |        0 | 1.12689e+06 |          0 | 2.03598e+06 |          0 |
+------------+--------+---------+----------+----------+-------------+------------+-------------+------------+
|          4 |   -4.5 |  12.625 |   416667 |        0 | 1.12689e+06 |          0 | 2.03598e+06 |          0 |
+------------+--------+---------+----------+----------+-------------+------------+-------------+------------+
|          5 |   -4.5 | -12.625 |   416667 |        0 | 1.12689e+06 |          0 | 2.03598e+06 |          0 |
+------------+--------+---------+----------+----------+-------------+------------+-------------+------------+
|          6 |   -4.5 |  17.125 |   416667 |        0 | 1.12689e+06 |          0 | 2.03598e+06 |          0 |
+------------+--------+---------+----------+----------+-------------+------------+-------------+------------+
|          7 |   -4.5 | -17.125 |   416667 |        0 | 1.12689e+06 |          0 | 2.03598e+06 |          0 |
+------------+--------+---------+----------+----------+-------------+------------+-------------+------------+
|          8 |      0 |   3.625 |   416667 |        0 |      416667 |          0 | 2.03598e+06 |          0 |
+------------+--------+---------+----------+----------+-------------+------------+-------------+------------+
|          9 |      0 |  -3.625 |   416667 |        0 |      416667 |          0 | 2.03598e+06 |          0 |
+------------+--------+---------+----------+----------+-------------+------------+-------------+------------+
|         10 |      0 |   8.125 |   416667 |        0 |      416667 |          0 | 2.03598e+06 |          0 |
+------------+--------+---------+----------+----------+-------------+------------+-------------+------------+
|         11 |      0 |  -8.125 |   416667 |        0 |      416667 |          0 | 2.03598e+06 |          0 |
+------------+--------+---------+----------+----------+-------------+------------+-------------+------------+
|         12 |      0 |  12.625 |   416667 |        0 |      416667 |          0 | 2.03598e+06 |          0 |
+------------+--------+---------+----------+----------+-------------+------------+-------------+------------+
|         13 |      0 | -12.625 |   416667 |        0 |      416667 |          0 | 2.03598e+06 |          0 |
+------------+--------+---------+----------+----------+-------------+------------+-------------+------------+
|         14 |      0 |  17.125 |   416667 |        0 |      416667 |          0 | 2.03598e+06 |          0 |
+------------+--------+---------+----------+----------+-------------+------------+-------------+------------+
|         15 |      0 | -17.125 |   416667 |        0 |      416667 |          0 | 2.03598e+06 |          0 |
+------------+--------+---------+----------+----------+-------------+------------+-------------+------------+
|         16 |    4.5 |   3.625 |   416667 |        0 |     -293561 |          0 | 2.03598e+06 |          0 |
+------------+--------+---------+----------+----------+-------------+------------+-------------+------------+
|         17 |    4.5 |  -3.625 |   416667 |        0 |     -293561 |          0 | 2.03598e+06 |          0 |
+------------+--------+---------+----------+----------+-------------+------------+-------------+------------+
|         18 |    4.5 |   8.125 |   416667 |        0 |     -293561 |          0 | 2.03598e+06 |          0 |
+------------+--------+---------+----------+----------+-------------+------------+-------------+------------+
|         19 |    4.5 |  -8.125 |   416667 |        0 |     -293561 |          0 | 2.03598e+06 |          0 |
+------------+--------+---------+----------+----------+-------------+------------+-------------+------------+
|         20 |    4.5 |  12.625 |   416667 |        0 |     -293561 |          0 | 2.03598e+06 |          0 |
+------------+--------+---------+----------+----------+-------------+------------+-------------+------------+
|         21 |    4.5 | -12.625 |   416667 |        0 |     -293561 |          0 | 2.03598e+06 |          0 |
+------------+--------+---------+----------+----------+-------------+------------+-------------+------------+
|         22 |    4.5 |  17.125 |   416667 |        0 |     -293561 |          0 | 2.03598e+06 |          0 |
+------------+--------+---------+----------+----------+-------------+------------+-------------+------------+
|         23 |    4.5 | -17.125 |   416667 |        0 |     -293561 |          0 | 2.03598e+06 |          0 |
+------------+--------+---------+----------+----------+-------------+------------+-------------+------------+

## 统计分析

### 位移和转角统计
+--------------+---------+---------+----------+-----------+-----------+-----------+
|  Statistics  |   Ux(m) |   Uy(m) |    Uz(m) |   Sx(rad) |   Sy(rad) |   Sz(rad) |
+==============+=========+=========+==========+===========+===========+===========+
|     Max      | 2.17493 |       0 | 0.794076 |         0 |  0.111215 |         0 |
+--------------+---------+---------+----------+-----------+-----------+-----------+
|     Min      | 2.17493 |       0 | -0.20686 |         0 |  0.111215 |         0 |
+--------------+---------+---------+----------+-----------+-----------+-----------+
|     Avg      | 2.17493 |       0 | 0.293608 |         0 |  0.111215 |         0 |
+--------------+---------+---------+----------+-----------+-----------+-----------+

### 力和弯矩统计信息
+--------------+----------+----------+-------------+------------+-------------+------------+
|  Statistics  |   Nx(kN) |   Ny(kN) |      Nz(kN) |   Mx(kN·m) |    My(kN·m) |   Mz(kN·m) |
+==============+==========+==========+=============+============+=============+============+
|     Max      |   416667 |        0 | 1.12689e+06 |          0 | 2.03598e+06 |          0 |
+--------------+----------+----------+-------------+------------+-------------+------------+
|     Min      |   416667 |        0 |     -293561 |          0 | 2.03598e+06 |          0 |
+--------------+----------+----------+-------------+------------+-------------+------------+
|     Avg      |   416667 |        0 |      416667 |          0 | 2.03598e+06 |          0 |
+--------------+----------+----------+-------------+------------+-------------+------------+


最不利单桩内力计算结果：
------------------------------------------------------------
最大轴向力: 1126894.70 kN (桩号: 1)
最大X向侧力:  416666.67 kN (桩号: 1)
最大Y向侧力:       0.00 kN (桩号: 0)
最大X向弯矩:       0.00 kN·m (桩号: 0)
最大Y向弯矩: 2035982.56 kN·m (桩号: 1)
最大扭矩:       0.00 kN·m (桩号: 0)
------------------------------------------------------------
沿桩身最大弯矩: 2035982.56 kN·m
  桩号: 1
  位置: Z =   0.00 m
  主要方向: Y
沿桩身最大剪力:  416666.67 kN
  桩号: 1
  位置: Z =   0.00 m
  主要方向: X
------------------------------------------------------------
综合最不利桩 (桩号: 1):
  轴向力 NZ: 1126894.70 kN
  侧向力 NX:  416666.67 kN
  侧向力 NY:       0.00 kN
  弯矩 MX:       0.00 kN·m
  弯矩 MY: 2035982.56 kN·m
  扭矩 MZ:       0.00 kN·m
------------------------------------------------------------